From: "John Spencer" <maillist-util-linux@barfooze.de>
Date: Mon Oct 19 14:23:58 +0000
Subject: [PATCH] fix invalid use of putp()

according to documentation, putp() is supposed to be only used with:
> ... a terminfo string variable or the return value from tparm, tgetstr, or
> tgoto.

for arbitrary strings, standard libc output routines need to be used.

this caused strings like "11 12 13"... to be truncated to "12 13" in the output
when built with netbsd-curses.

--- util-linux-2.23.1.org/misc-utils/cal.c
+++ util-linux-2.23.1/misc-utils/cal.c
@@ -87,11 +87,6 @@
     setupterm((char*)term, fildes, errret);
 }
 
-static void
-my_putstring(char *s) {
-     putp(s);
-}
-
 static const char *
 my_tgetstr(char *s __attribute__ ((__unused__)), char *ss) {
     const char* ret = tigetstr(ss);
@@ -114,11 +109,6 @@
     *errret = tgetent(tcbuffer, term);
 }
 
-static void
-my_putstring(char *s) {
-     tputs (s, 1, putchar);
-}
-
 static const char *
 my_tgetstr(char *s, char *ss __attribute__ ((__unused__))) {
     const char* ret = tgetstr(s, &strbuf);
@@ -128,14 +118,12 @@
 	return ret;
 }
 
-#else /* ! (HAVE_LIBTERMCAP || HAVE_LIBNCURSES || HAVE_LIBNCURSESW) */
+#endif
 
 static void
 my_putstring(char *s) {
      fputs(s, stdout);
 }
-
-#endif
 
 
 const char	*term="";
